shared/src/lib.rs

```diff
@@ -560,120 +560,11 @@ pub enum CredentialTransfer {
-    fn default() -> Self {
-        EdhocMessageBuffer {
-            content: [0; MAX_MESSAGE_SIZE_LEN],
-            len: 0,
-        }
-    }
-}
-
-impl EdhocMessageBuffer {
-    pub fn new() -> Self {
-        EdhocMessageBuffer {
-            content: [0u8; MAX_MESSAGE_SIZE_LEN],
-            len: 0,
-        }
-    }
-
-    pub fn new_from_slice(slice: &[u8]) -> Result<Self, MessageBufferError> {
-        let mut buffer = Self::new();
-        if buffer.fill_with_slice(slice).is_ok() {
-            Ok(buffer)
-        } else {
-            Err(MessageBufferError::SliceTooLong)
-        }
-    }
-
-    pub fn get(self, index: usize) -> Option<u8> {
-        self.content.get(index).copied()
-    }
-
-    pub fn push(&mut self, item: u8) -> Result<(), MessageBufferError> {
-        if self.len < self.content.len() {
-            self.content[self.len] = item;
-            self.len += 1;
-            Ok(())
-        } else {
-            Err(MessageBufferError::BufferAlreadyFull)
-        }
-    }
-
-    pub fn get_slice(&self, start: usize, len: usize) -> Option<&[u8]> {
-        self.content.get(start..start + len)
-    }
-
-    pub fn as_slice(&self) -> &[u8] {
-        &self.content[0..self.len]
-    }
-
-    pub fn fill_with_slice(&mut self, slice: &[u8]) -> Result<(), MessageBufferError> {
-        if slice.len() <= self.content.len() {
-            self.len = slice.len();
-            self.content[..self.len].copy_from_slice(slice);
-            Ok(())
-        } else {
-            Err(MessageBufferError::SliceTooLong)
-        }
-    }
-
-    pub fn extend_from_slice(&mut self, slice: &[u8]) -> Result<(), MessageBufferError> {
-        if self.len + slice.len() <= self.content.len() {
-            self.content[self.len..self.len + slice.len()].copy_from_slice(slice);
-            self.len += slice.len();
-            Ok(())
-        } else {
-            Err(MessageBufferError::SliceTooLong)
-        }
-    }
-
-    pub fn from_hex(hex: &str) -> Self {
-        let mut buffer = EdhocMessageBuffer::new();
-        buffer.len = hex.len() / 2;
-        for (i, chunk) in hex.as_bytes().chunks(2).enumerate() {
-            let chunk_str = core::str::from_utf8(chunk).unwrap();
-            buffer.content[i] = u8::from_str_radix(chunk_str, 16).unwrap();
-        }
-        buffer
-    }
-}
+#[deprecated]
+pub type MessageBufferError = buffer::EdhocBufferError;
 
-impl TryInto<EdhocMessageBuffer> for &[u8] {
-    type Error = ();
-
-    fn try_into(self) -> Result<EdhocMessageBuffer, Self::Error> {
-        let mut buffer = [0u8; MAX_MESSAGE_SIZE_LEN];
-        if self.len() <= buffer.len() {
-            buffer[..self.len()].copy_from_slice(self);
-
-            Ok(EdhocMessageBuffer {
-                content: buffer,
-                len: self.len(),
-            })
-        } else {
-            Err(())
-        }
-    }
-}
+/// An [`EdhocBuffer`] used for messages (and, transitionally, other components).
```

Comment:
Should I elaborate on the "transitionally" here? (I think that'll become clear soon enough as we start not using a full EdhocMessageBuffer for everything, going forward from here).

Reply:
Yes; it's really up next, but doing that will just need a tad more testing compared to this (which is almost a binaries-should-be-identical PR).